crates/common/src/runtime_client.rs

```diff
@@ -128,7 +128,9 @@ impl RuntimeClient {
     async fn connect(&self) -> Result<InnerClient, RuntimeClientError> {
         match self.url.scheme() {
             "http" | "https" => {
-                let mut client_builder = reqwest::Client::builder().timeout(self.timeout);
+                let mut client_builder = reqwest::Client::builder()
+                    .timeout(self.timeout)
+                    .tls_built_in_root_certs(self.url.scheme() == "https");
```

Comment:
right, this always defaults to true but we only need it if the provider URL is https.